app/lib/max_allowed_days_loader.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module MaxAllowedDaysLoader
+  DEFAULTS = {
+    account_scheduled_for_deletion: 15.days,
+    account_suspension: 90.days,
+    account_inactivity: 1.year,
+    contract_unpaid_time: 6.months
+  }.freeze
+
+  module_function
+
+  def load_account_scheduled_for_deletion
+    config[:account_scheduled_for_deletion]
+  end
+
+  def load_account_suspension
+    config[:account_suspension]
+  end
+
+  def load_account_inactivity
+    config[:account_inactivity]
+  end
+
+  def load_contract_unpaid_time
+    config[:contract_unpaid_time]
+  end
+
+  def config
+    @config ||= load_config
+  end
+
+  def load_config
+    initial_config = (ThreeScale.config.max_allowed_days || {})
+    initial_config.each_with_object({}) do |(key, value), collection|
+      collection[key.to_sym] = value.public_send(:days) if valid_value?(value)
+    end.reverse_merge(DEFAULTS)
+  end
+
+  def valid_value?(value)
```

Comment:
Not sure if we need this. We have a bunch of settings already where pretty much any one of them could break if not properly set. I don't think we should type check this one. Maybe it's something we can leave for documentation.

Reply:
I think @hallelujah should give an opinion about this since he is the one who requested it:
https://github.com/3scale/porta/pull/555#discussion_r251485309
![image](https://user-images.githubusercontent.com/11318903/52269255-d97b6f80-293d-11e9-955d-cee827772fe6.png)

I asked him:
> Question about this: In the configuration yml, the value should be a string like `'6 months'` and we convert it to `6.months` in code... or should it be a number that represents the number of days?

And he said:
> number of days

Reply:
> where pretty much any one of them could break if not properly set

Here nothing should break if not properly set. It just picks the defaults instead 😄

Reply:
"Sane and safe defaults" does not mean type check. And "breaking if not properly set" is better than picking up the default instead, which leads to completely unexpected behavior.

Reply:
> "Sane and safe defaults" does not mean type check

You are right, but if I have to check that is not a negative number, then it is going to raise an error if it is the wrong type.

> And "breaking if not properly set" is better than picking up the default instead, which leads to completely unexpected behavior.

I disagree because this is in deployment time, which means that if it fails, it won't deploy at all... and this config shouldn't be that important like to don't deploy 😱 Maximum showing an error message or sending an email about it...

Reply:
Since it is very related, let's move this conversation to https://github.com/3scale/porta/pull/574#discussion_r253822994 so we don't discuss the same twice 😄